packages/ui/src/components/shadcn/ui/navigation-menu.tsx

```diff
@@ -7,20 +7,21 @@ import { cn } from '../../../lib/utils/cn'
 
 interface NavigationMenuProps
   extends React.ComponentPropsWithoutRef<typeof NavigationMenuPrimitive.Root> {
+  hasViewport?: boolean | undefined
   viewportClassName?: string | undefined
 }
 
 const NavigationMenu = React.forwardRef<
   React.ElementRef<typeof NavigationMenuPrimitive.Root>,
   NavigationMenuProps
->(({ className, viewportClassName, children, ...props }, ref) => (
+>(({ className, hasViewport = true, viewportClassName, children, ...props }, ref) => (
   <NavigationMenuPrimitive.Root
     ref={ref}
     className={cn('relative z-10 flex flex-1 items-center justify-center', className)}
     {...props}
   >
     {children}
-    <NavigationMenuViewport className={viewportClassName} />
+    {hasViewport && <NavigationMenuViewport className={viewportClassName} />}
```

Comment:
?

not sure what you're doing here. are we misusing the component?

Reply:
I couldn't find any other way to position the dropdown under the trigger apart from removing the NavigationMenuViewport.

It seems to be a common issue: https://github.com/shadcn-ui/ui/issues/94#issuecomment-1986921032

Reply:
looked in the radix docs / issues ?

Reply:
I believe I have

Reply:
this is solved now?